realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1016,10 +1016,19 @@ public void executeTransaction(Transaction transaction) {
             transaction.execute(this);
             commitTransaction();
         } catch (RuntimeException e) {
```

Comment:
I'm trying to think we we are making a difference between Runtime and Exceptions, and fail to come up with a sensible one. The only thing it does is masking the real error, which was also the complaint in the issue. IMO we should remove the wrapper RealmException, and just collapse these two. What do you think?